test/unit-tests/MultiMessageSender.t.sol

```diff
@@ -0,0 +1,375 @@
+        super.setUp();
+
+        vm.selectFork(fork[SRC_CHAIN_ID]);
+        sender = MultiMessageSender(contractAddress[SRC_CHAIN_ID]["MMA_SENDER"]);
+        receiver = contractAddress[DST_CHAIN_ID]["MMA_RECEIVER"];
+        gac = IGAC(contractAddress[SRC_CHAIN_ID]["GAC"]);
+        wormholeAdapterAddr = contractAddress[SRC_CHAIN_ID]["WORMHOLE_SENDER_ADAPTER"];
+        axelarAdapterAddr = contractAddress[SRC_CHAIN_ID]["AXELAR_SENDER_ADAPTER"];
+    }
+
+    /// @dev constructor
+    function test_constructor() public {
+        // checks existing setup
+        assertEq(address(sender.gac()), contractAddress[SRC_CHAIN_ID]["GAC"]);
+    }
+
+    /// @dev cannot be called with zero address GAC
+    function test_constructor_zero_address_input() public {
+        vm.expectRevert(Error.ZERO_ADDRESS_INPUT.selector);
+        new MultiMessageSender(address(0));
+    }
+
+    /// @dev perform remote call
+    function test_remote_call() public {
+        vm.startPrank(caller);
+
+        address[] memory senderAdapters = new address[](2);
+        senderAdapters[0] = wormholeAdapterAddr;
+        senderAdapters[1] = axelarAdapterAddr;
+
+        bool[] memory adapterSuccess = new bool[](2);
+        adapterSuccess[0] = true;
+        adapterSuccess[1] = true;
+
+        uint256 expiration = block.timestamp + gac.getMsgExpiryTime();
+
+        MessageLibrary.Message memory message = MessageLibrary.Message({
+            srcChainId: SRC_CHAIN_ID,
+            dstChainId: DST_CHAIN_ID,
+            target: address(42),
+            nonce: 1,
+            callData: bytes("42"),
+            expiration: expiration
+        });
+        bytes32 msgId = MessageLibrary.computeMsgId(message);
+
+        uint256 fee = sender.estimateTotalMessageFee(DST_CHAIN_ID, receiver, address(42), bytes("42"));
+
+        vm.expectEmit(true, true, true, true, address(sender));
+        emit MultiMessageMsgSent(
+            msgId, 1, DST_CHAIN_ID, address(42), bytes("42"), expiration, senderAdapters, adapterSuccess
+        );
+
+        sender.remoteCall{value: fee}(DST_CHAIN_ID, address(42), bytes("42"));
+
+        assertEq(sender.nonce(), 1);
+    }
+
+    /// @dev perform remote call, checking for refund
+    function test_remote_call_refund() public {
+        vm.startPrank(caller);
+
+        uint256 expiration = block.timestamp + gac.getMsgExpiryTime();
+
+        MessageLibrary.Message memory message = MessageLibrary.Message({
+            srcChainId: SRC_CHAIN_ID,
+            dstChainId: DST_CHAIN_ID,
+            target: address(42),
+            nonce: 0,
+            callData: bytes("42"),
+            expiration: expiration
+        });
+        bytes32 msgId = MessageLibrary.computeMsgId(message);
+
+        uint256 balanceBefore = gac.getRefundAddress().balance;
+        sender.remoteCall{value: 2 ether}(DST_CHAIN_ID, address(42), bytes("42"));
+
+        // TODO: Maybe try to make this precise
+        assertEq(address(sender).balance, 0);
```

Comment:
Done.

Great, previously I thought `assertEq` would not work due to gas but now I realized we are testing in a contract so it's easier to be precise.

The current implementation always refunds the sender's balance, which is technically incorrect in case someone sends ETH directly to the sender. We can address that later.